os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -1809,3 +1836,183 @@ struct i2s_dev_s *amebasmart_i2s_initialize(uint16_t port)
+ *   Notify the driver of new power state. This callback is  called after
+ *   all drivers have had the opportunity to prepare for the new power state.
+ *
+ * Input Parameters:
+ *
+ *    cb - Returned to the driver. The driver version of the callback
+ *         structure may include additional, driver-specific state data at
+ *         the end of the structure.
+ *
+ *    pmstate - Identifies the new PM state
+ *
+ * Returned Value:
+ *   None - The driver already agreed to transition to the low power
+ *   consumption state when when it returned OK to the prepare() call.
+ *
+ *
+ ****************************************************************************/
+#ifdef CONFIG_PM
+static void amebasmart_i2s_pmnotify(FAR struct pm_callback_s *cb, int domain,
+                                   enum pm_state_e pmstate)
+{
+	switch (pmstate)
+	{
+		case PM_NORMAL:
+		case PM_IDLE:
+		case PM_STANDBY:
+		case PM_SLEEP:
+			break;
+		default:
+			break;
+	}
+
+	return;
+}
+#endif
+
+/****************************************************************************
+ * Name: amebasmart_i2s_pmprepare
+ *
+ * Description:
+ *   Request the driver to prepare for a new power state. This is a warning
+ *   that the system is about to enter into a new power state. The driver
+ *   should begin whatever operations that may be required to enter power
+ *   state. The driver may abort the state change mode by returning a
+ *   non-zero value from the callback function.
+ *
+ * Input Parameters:
+ *
+ *    cb - Returned to the driver. The driver version of the callback
+ *         structure may include additional, driver-specific state data at
+ *         the end of the structure.
+ *
+ *    pmstate - Identifies the new PM state
+ *
+ * Returned Value:
+ *   Zero - (OK) means the event was successfully processed and that the
+ *          driver is prepared for the PM state change.
+ *
+ *   Non-zero - means that the driver is not prepared to perform the tasks
+ *              needed achieve this power setting and will cause the state
+ *              change to be aborted. NOTE: The prepare() method will also
+ *              be called when reverting from lower back to higher power
+ *              consumption modes (say because another driver refused a
+ *              lower power state change). Drivers are not permitted to
+ *              return non-zero values when reverting back to higher power
+ *              consumption modes!
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_PM
+static int amebasmart_i2s_pmprepare(FAR struct pm_callback_s *cb, int domain, enum pm_state_e pmstate)
+{
+	switch (pmstate)
+	{
+		case PM_NORMAL:
+		case PM_IDLE:
+		case PM_STANDBY:
+		case PM_SLEEP:
+			break;
+		default:
+			break;
+	}
+
+	return OK;
+}
+#endif
+
+
+#ifdef CONFIG_PM
+void i2s_pminitialize(void)
+{
+	int ret;
+	/* Register to receive power management callbacks */
+	ret = pm_register(&g_i2spm.pm_cb);
```

Comment:
If you do not use this, Could you please remove it?
This can cause unnecessary memory and sleep-wakeup time to increase.

Reply:
Yes, I understand your concern~ Removed accordingly